pkgs/development/python-modules/lightgbm/default.nix

```diff
@@ -28,16 +34,25 @@ buildPythonPackage rec {
 
   dontUseCmakeConfigure = true;
 
-  buildInputs = lib.optionals stdenv.cc.isClang [
+  buildInputs = (optionals stdenv.cc.isClang [
     llvmPackages.openmp
-  ];
+  ]) ++ (optionals gpuSupport [
+    boost
+    ocl-icd
+    opencl-headers
+  ]);
 
   propagatedBuildInputs = [
     numpy
     scipy
     scikit-learn
   ];
 
+  buildPhase = concatStringsSep " " ([
+    python.pythonForBuild.interpreter
+    "setup.py bdist_wheel"
+  ] ++ optionals gpuSupport ["--gpu"]);
```

Comment:
Please use the appropriate flag instead

Reply:
Can you explain what you mean by the appropriate flag? `--gpu` is the flag for building with GPU support.

Reply:
Already tried those. None of them allow you to pass `--gpu` as a flag at the end.
For example, adding `setupPyBuildFlags = optionals gpuSupport [ "--gpu" ];` will lead to this command getting run in the build phase:
```
python3.10 nix_run_setup build_ext --gpu bdist_wheel
```
This throws `error: option --gpu not recognized`

From https://github.com/NixOS/nixpkgs/blob/8dc1ebbfa66faa822140c07dc7b146f810e67109/pkgs/development/interpreters/python/hooks/setuptools-build-hook.sh#L20 it doesn't look like there is any way to pass the flag after bdist_wheel.

Reply:
Hmmm, I see, can we just patch the source code to enable the flag? Vendoring bdist_wheel here will probably break in the far future when buildPythonPackage is changed.

Reply:
As mentioned in the comment about the package variants, there will be multiple combinations of flags for different versions of the package. Patching the code will get very messy very fast. Besides, any changes to buildPythonPackage significant enough to break this would probably break other packages too. There are 83 packages in nixpkgs that directly invoke setup.py
```
[illustris@desktop:~/src/nixpkgs]$ grep -rIn python.pythonForBuild.interpreter | wc -l
83
```

A cleaner solution to this problem is to add another option, maybe `setupPyBdistFlags`, to add flags to bdist_wheel. But I'll leave that to maintainers of the broader python ecosystem such as yourself.

Reply:
If we don't need it, we probably will not add it.

Reply:
This is one example of a package that needs it. I'm not sure how many of the 83 packages that overrode the buildPhase like this one did it because of this limitation with buildPythonPackage. Either way, if there are no plans to add this option, overriding buildPhase the way it's done right now is the cleanest solution.